### ISSUE TITLE
Few tests fail

### ISSUE THREAD

**Issue by yurivict** [fmtlib/fmt#3512]
The log isn't clear on what tests fail, so posting the whole log:
```
      Start  1: args-test
 1/20 Test  #1: args-test ........................   Passed    0.00 sec
      Start  2: assert-test
 2/20 Test  #2: assert-test ......................   Passed    0.00 sec
      Start  3: chrono-test
 3/20 Test  #3: chrono-test ......................   Passed    0.12 sec
      Start  4: color-test
 4/20 Test  #4: color-test .......................   Passed    0.00 sec
      Start  5: core-test
 5/20 Test  #5: core-test ........................   Passed    0.00 sec
      Start  6: gtest-extra-test
 6/20 Test  #6: gtest-extra-test .................   Passed    0.00 sec
      Start  7: format-test
 7/20 Test  #7: format-test ......................***Failed    0.03 sec
[==========] Running 114 tests from 6 test suites.
[----------] Global test environment set-up.
[----------] 5 tests from uint128_test
[ RUN      ] uint128_test.ctor
[       OK ] uint128_test.ctor (0 ms)
[ RUN      ] uint128_test.shift
[       OK ] uint128_test.shift (0 ms)
[ RUN      ] uint128_test.minus
[       OK ] uint128_test.minus (0 ms)
[ RUN      ] uint128_test.plus_assign
[       OK ] uint128_test.plus_assign (0 ms)
[ RUN      ] uint128_test.multiply
[       OK ] uint128_test.multiply (0 ms)
[----------] 5 tests from uint128_test (0 ms total)

[----------] 2 tests from float_test
[ RUN      ] float_test.isfinite
[       OK ] float_test.isfinite (0 ms)
[ RUN      ] float_test.isnan
[       OK ] float_test.isnan (0 ms)
[----------] 2 tests from float_test (0 ms total)

[----------] 9 tests from util_test
[ RUN      ] util_test.bit_cast
[       OK ] util_test.bit_cast (0 ms)
[ RUN      ] util_test.increment
[       OK ] util_test.increment (0 ms)
[ RUN      ] util_test.parse_nonnegative_int
[       OK ] util_test.parse_nonnegative_int (0 ms)
[ RUN      ] util_test.utf8_to_utf16
[       OK ] util_test.utf8_to_utf16 (0 ms)
[ RUN      ] util_test.utf8_to_utf16_empty_string
[       OK ] util_test.utf8_to_utf16_empty_string (0 ms)
[ RUN      ] util_test.allocator_ref
[       OK ] util_test.allocator_ref (0 ms)
[ RUN      ] util_test.format_system_error
warning: std::allocator allocates 9223372036854775807 chars
[       OK ] util_test.format_system_error (0 ms)
[ RUN      ] util_test.system_error
[       OK ] util_test.system_error (0 ms)
[ RUN      ] util_test.report_system_error
[       OK ] util_test.report_system_error (0 ms)
[----------] 9 tests from util_test (0 ms total)

[----------] 9 tests from memory_buffer_test
[ RUN      ] memory_buffer_test.ctor
[       OK ] memory_buffer_test.ctor (0 ms)
[ RUN      ] memory_buffer_test.move_ctor_inline_buffer
[       OK ] memory_buffer_test.move_ctor_inline_buffer (0 ms)
[ RUN      ] memory_buffer_test.move_ctor_dynamic_buffer
[       OK ] memory_buffer_test.move_ctor_dynamic_buffer (0 ms)
[ RUN      ] memory_buffer_test.move_assignment
[       OK ] memory_buffer_test.move_assignment (0 ms)
[ RUN      ] memory_buffer_test.grow
[       OK ] memory_buffer_test.grow (0 ms)
[ RUN      ] memory_buffer_test.allocator
[       OK ] memory_buffer_test.allocator (0 ms)
[ RUN      ] memory_buffer_test.exception_in_deallocate
[       OK ] memory_buffer_test.exception_in_deallocate (0 ms)
[ RUN      ] memory_buffer_test.max_size_allocator
[       OK ] memory_buffer_test.max_size_allocator (0 ms)
[ RUN      ] memory_buffer_test.max_size_allocator_overflow
[       OK ] memory_buffer_test.max_size_allocator_overflow (0 ms)
[----------] 9 tests from memory_buffer_test (0 ms total)

[----------] 87 tests from format_test
[ RUN      ] format_test.escape
[       OK ] format_test.escape (0 ms)
[ RUN      ] format_test.unmatched_braces
[       OK ] format_test.unmatched_braces (0 ms)
[ RUN      ] format_test.no_args
[       OK ] format_test.no_args (0 ms)
[ RUN      ] format_test.args_in_different_positions
[       OK ] format_test.args_in_different_positions (0 ms)
[ RUN      ] format_test.arg_errors
[       OK ] format_test.arg_errors (0 ms)
[ RUN      ] format_test.many_args
[       OK ] format_test.many_args (0 ms)
[ RUN      ] format_test.named_arg
[       OK ] format_test.named_arg (0 ms)
[ RUN      ] format_test.auto_arg_index
[       OK ] format_test.auto_arg_index (0 ms)
[ RUN      ] format_test.empty_specs
[       OK ] format_test.empty_specs (0 ms)
[ RUN      ] format_test.left_align
[       OK ] format_test.left_align (0 ms)
[ RUN      ] format_test.right_align
[       OK ] format_test.right_align (0 ms)
[ RUN      ] format_test.center_align
[       OK ] format_test.center_align (0 ms)
[ RUN      ] format_test.fill
[       OK ] format_test.fill (0 ms)
[ RUN      ] format_test.plus_sign
[       OK ] format_test.plus_sign (0 ms)
[ RUN      ] format_test.minus_sign
[       OK ] format_test.minus_sign (0 ms)
[ RUN      ] format_test.space_sign
[       OK ] format_test.space_sign (0 ms)
[ RUN      ] format_test.hash_flag
[       OK ] format_test.hash_flag (0 ms)
[ RUN      ] format_test.zero_flag
[       OK ] format_test.zero_flag (0 ms)
[ RUN      ] format_test.zero_flag_and_align
[       OK ] format_test.zero_flag_and_align (0 ms)
[ RUN      ] format_test.width
[       OK ] format_test.width (0 ms)
[ RUN      ] format_test.runtime_width
[       OK ] format_test.runtime_width (0 ms)
[ RUN      ] format_test.precision
[       OK ] format_test.precision (0 ms)
[ RUN      ] format_test.runtime_precision
[       OK ] format_test.runtime_precision (0 ms)
[ RUN      ] format_test.format_bool
[       OK ] format_test.format_bool (0 ms)
[ RUN      ] format_test.format_short
[       OK ] format_test.format_short (0 ms)
[ RUN      ] format_test.format_int
[       OK ] format_test.format_int (4 ms)
[ RUN      ] format_test.format_bin
[       OK ] format_test.format_bin (0 ms)
[ RUN      ] format_test.format_dec
[       OK ] format_test.format_dec (0 ms)
[ RUN      ] format_test.format_hex
[       OK ] format_test.format_hex (0 ms)
[ RUN      ] format_test.format_oct
[       OK ] format_test.format_oct (0 ms)
[ RUN      ] format_test.format_int_locale
[       OK ] format_test.format_int_locale (0 ms)
[ RUN      ] format_test.format_float
[       OK ] format_test.format_float (0 ms)
[ RUN      ] format_test.format_double
/usr/ports/devel/libfmt/work/fmt-10.0.0/test/format-test.cc:1383: Failure
Expected equality of these values:
  fmt::format("{:a}", ld)
    Which is: "0x8p-16385"
  buffer
    Which is: "0x1p-16382"
/usr/ports/devel/libfmt/work/fmt-10.0.0/test/format-test.cc:1387: Failure
Expected equality of these values:
  fmt::format("{:a}", ld)
    Which is: "0xf.fffffffffffffffp+16380"
  buffer
    Which is: "0x1.fffffffffffffffep+16383"
[  FAILED  ] format_test.format_double (5 ms)
[ RUN      ] format_test.precision_rounding
[       OK ] format_test.precision_rounding (0 ms)
[ RUN      ] format_test.prettify_float
[       OK ] format_test.prettify_float (0 ms)
[ RUN      ] format_test.format_nan
[       OK ] format_test.format_nan (0 ms)
[ RUN      ] format_test.format_infinity
[       OK ] format_test.format_infinity (0 ms)
[ RUN      ] format_test.format_long_double
/usr/ports/devel/libfmt/work/fmt-10.0.0/test/format-test.cc:1499: Failure
Expected equality of these values:
  buffer
    Which is: "0x1.a7ae147ae147ae14p+1"
  fmt::format("{:a}", ld)
    Which is: "0xd.3d70a3d70a3d70ap-2"
[  FAILED  ] format_test.format_long_double (0 ms)
[ RUN      ] format_test.format_char
[       OK ] format_test.format_char (4 ms)
[ RUN      ] format_test.format_volatile_char
[       OK ] format_test.format_volatile_char (0 ms)
[ RUN      ] format_test.format_unsigned_char
[       OK ] format_test.format_unsigned_char (0 ms)
[ RUN      ] format_test.format_cstring
[       OK ] format_test.format_cstring (4 ms)
[ RUN      ] format_test.format_pointer
[       OK ] format_test.format_pointer (4 ms)
[ RUN      ] format_test.write_uintptr_fallback
[       OK ] format_test.write_uintptr_fallback (0 ms)
[ RUN      ] format_test.format_enum_class
[       OK ] format_test.format_enum_class (0 ms)
[ RUN      ] format_test.format_string
[       OK ] format_test.format_string (0 ms)
[ RUN      ] format_test.format_string_view
[       OK ] format_test.format_string_view (0 ms)
[ RUN      ] format_test.format_std_string_view
[       OK ] format_test.format_std_string_view (0 ms)
[ RUN      ] format_test.format_explicitly_convertible_to_std_string_view
[       OK ] format_test.format_explicitly_convertible_to_std_string_view (0 ms)
[ RUN      ] format_test.format_custom
[       OK ] format_test.format_custom (0 ms)
[ RUN      ] format_test.format_to_custom
[       OK ] format_test.format_to_custom (0 ms)
[ RUN      ] format_test.format_string_from_speed_test
[       OK ] format_test.format_string_from_speed_test (0 ms)
[ RUN      ] format_test.format_examples
[       OK ] format_test.format_examples (0 ms)
[ RUN      ] format_test.print
[       OK ] format_test.print (0 ms)
[ RUN      ] format_test.variadic
[       OK ] format_test.variadic (0 ms)
[ RUN      ] format_test.dynamic
[       OK ] format_test.dynamic (0 ms)
[ RUN      ] format_test.bytes
[       OK ] format_test.bytes (0 ms)
[ RUN      ] format_test.group_digits_view
[       OK ] format_test.group_digits_view (0 ms)
[ RUN      ] format_test.join
[       OK ] format_test.join (0 ms)
[ RUN      ] format_test.format_message_example
[       OK ] format_test.format_message_example (0 ms)
[ RUN      ] format_test.unpacked_args
[       OK ] format_test.unpacked_args (0 ms)
[ RUN      ] format_test.compile_time_string
[       OK ] format_test.compile_time_string (0 ms)
[ RUN      ] format_test.custom_format_compile_time_string
[       OK ] format_test.custom_format_compile_time_string (0 ms)
[ RUN      ] format_test.named_arg_udl
[       OK ] format_test.named_arg_udl (0 ms)
[ RUN      ] format_test.enum
[       OK ] format_test.enum (0 ms)
[ RUN      ] format_test.formatter_not_specialized
[       OK ] format_test.formatter_not_specialized (0 ms)
[ RUN      ] format_test.strong_enum
[       OK ] format_test.strong_enum (0 ms)
[ RUN      ] format_test.non_null_terminated_format_string
[       OK ] format_test.non_null_terminated_format_string (0 ms)
[ RUN      ] format_test.to_string
[       OK ] format_test.to_string (0 ms)
[ RUN      ] format_test.output_iterators
[       OK ] format_test.output_iterators (0 ms)
[ RUN      ] format_test.formatted_size
[       OK ] format_test.formatted_size (0 ms)
[ RUN      ] format_test.format_to_no_args
[       OK ] format_test.format_to_no_args (0 ms)
[ RUN      ] format_test.format_to
[       OK ] format_test.format_to (0 ms)
[ RUN      ] format_test.format_to_memory_buffer
[       OK ] format_test.format_to_memory_buffer (0 ms)
[ RUN      ] format_test.format_to_vector
[       OK ] format_test.format_to_vector (0 ms)
[ RUN      ] format_test.format_to_propagates_exceptions
[       OK ] format_test.format_to_propagates_exceptions (0 ms)
[ RUN      ] format_test.format_to_n
[       OK ] format_test.format_to_n (0 ms)
[ RUN      ] format_test.format_to_n_output_iterator
[       OK ] format_test.format_to_n_output_iterator (0 ms)
[ RUN      ] format_test.vformat_to
[       OK ] format_test.vformat_to (0 ms)
[ RUN      ] format_test.char_traits_is_not_ambiguous
[       OK ] format_test.char_traits_is_not_ambiguous (0 ms)
[ RUN      ] format_test.back_insert_slicing
[       OK ] format_test.back_insert_slicing (0 ms)
[ RUN      ] format_test.format_as
[       OK ] format_test.format_as (0 ms)
[ RUN      ] format_test.test_formatters_enabled
[       OK ] format_test.test_formatters_enabled (0 ms)
[ RUN      ] format_test.format_facet
[       OK ] format_test.format_facet (0 ms)
[ RUN      ] format_test.format_facet_separator
[       OK ] format_test.format_facet_separator (0 ms)
[ RUN      ] format_test.format_facet_grouping
[       OK ] format_test.format_facet_grouping (0 ms)
[ RUN      ] format_test.format_named_arg_with_locale
[       OK ] format_test.format_named_arg_with_locale (0 ms)
[----------] 87 tests from format_test (29 ms total)

[----------] 2 tests from format_int_test
[ RUN      ] format_int_test.data
[       OK ] format_int_test.data (0 ms)
[ RUN      ] format_int_test.format_int
[       OK ] format_int_test.format_int (0 ms)
[----------] 2 tests from format_int_test (0 ms total)

[----------] Global test environment tear-down
[==========] 114 tests from 6 test suites ran. (30 ms total)
[  PASSED  ] 112 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] format_test.format_double
[  FAILED  ] format_test.format_long_double

 2 FAILED TESTS

      Start  8: format-impl-test
 8/20 Test  #8: format-impl-test .................   Passed    0.05 sec
      Start  9: ostream-test
 9/20 Test  #9: ostream-test .....................   Passed    0.00 sec
      Start 10: compile-test
10/20 Test #10: compile-test .....................   Passed    0.00 sec
      Start 11: compile-fp-test
11/20 Test #11: compile-fp-test ..................   Passed    0.00 sec
      Start 12: printf-test
12/20 Test #12: printf-test ......................   Passed    0.01 sec
      Start 13: ranges-test
13/20 Test #13: ranges-test ......................   Passed    0.00 sec
      Start 14: scan-test
14/20 Test #14: scan-test ........................   Passed    0.00 sec
      Start 15: std-test
15/20 Test #15: std-test .........................   Passed    0.01 sec
      Start 16: unicode-test
16/20 Test #16: unicode-test .....................   Passed    0.00 sec
      Start 17: xchar-test
17/20 Test #17: xchar-test .......................   Passed    0.01 sec
      Start 18: enforce-checks-test
18/20 Test #18: enforce-checks-test ..............   Passed    0.00 sec
      Start 19: posix-mock-test
19/20 Test #19: posix-mock-test ..................   Passed    0.00 sec
      Start 20: os-test
20/20 Test #20: os-test ..........................   Passed    0.00 sec

95% tests passed, 1 tests failed out of 20

Total Test time (real) =   0.29 sec

The following tests FAILED:
	  7 - format-test (Failed)
```

Version: 10.0.0
FreeBSD 13.2

**Comment by phprus** [comment]
Previous issue: #3433
Fix: #3434

**Comment by phprus** [comment]
This and #3433 issue is bug in tests. Not in fmt code.
